### PR TITLE
Add missing sudo in BUILDING and README

### PR DESCRIPTION
**Pull request by Maijin**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [x] I've updated the documentation and the [rizin book](https://github.com/rizinorg/rizinbook) with the relevant information (if needed)

**Detailed description**

Add missing sudo in BUILDING and README.

### CODE REVIEW

File: BUILDING.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ distribution in `/usr`.
 ```
 $ meson build
 $ ninja -C build # or `meson compile -C build`
-$ sudo ninja -C build install # or `meson install -C build`
+$ ninja -C build install # or `meson install -C build`
```
Comment (ret2libc) [inline BUILDING.md]: Why unnecessary? I think it is necessary to have sudo privileges to install in /usr/local. What system are you using?

**Review by ret2libc — CHANGES_REQUESTED** [review]

File: BUILDING.md (hunk 1 above)
Comment (ret2libc) [reply BUILDING.md]: Actually, could you add `sudo` also to the instructions in README.md? On most systems `sudo` will be needed to install to /usr/local

Comment (Maijin) [reply BUILDING.md]: on Mac OS it's not required but I'm fine changing both to sudo just let me know

Comment (wargio) [reply BUILDING.md]: i do agree with ret2libc. the only one that should not have sudo is when you install the binary locally.

Comment (Maijin) [reply BUILDING.md]: ok let me change everywhere then

File: BUILDING.md [hunk 2]
```diff
@@ -164,7 +164,7 @@ If Rizin was installed using `meson`, you can run the following command from the
 same build directory where you had previously installed Rizin:
 
 ```
-$ ninja -C uninstall
+$ sudo ninja -C uninstall
```
Comment (ret2libc) [inline BUILDING.md]:
```suggestion
$ sudo ninja -C uninstall # `sudo` may not be required based on how you configured the `build` directory with meson the first time```

File: BUILDING.md [hunk 3]
```diff
@@ -184,7 +184,7 @@ codebase and its submodules to the latest version.
 To re-build Rizin after you have updated your source code, you can use:
 ```
 $ ninja -C build # or `meson compile -C build`
-$ sudo ninja -C build install # or `meson install -C build`
+$ sudo ninja -C build install # or `sudo meson install -C build`
```
Comment (ret2libc) [inline BUILDING.md]:
```suggestion
$ sudo ninja -C build install # or `sudo meson install -C build`. `sudo` may not be required based on how you configured the `build` directory with meson the first time
```

**Review by ITAYC0HEN — APPROVED** [review]

**Review by ret2libc — APPROVED** [review]

**Comment by ret2libc** [comment]
Thanks @Maijin ! I added some small notes about the fact that `sudo` may not be required in some cases. Just to warn users.

**Pull request merged**